pkg/klusterlet/clusterinfo/clusterinfo_controller.go

```diff
@@ -388,6 +399,47 @@ func (r *ClusterInfoReconciler) getDistributionInfo() (clusterv1beta1.Distributi
 	return distributionInfo, nil
 }
 
+func distributionInfoUpdated(old, new *clusterv1beta1.DistributionInfo) bool {
+	switch new.Type {
+	case clusterv1beta1.DistributionTypeOCP:
+		return ocpDistributionInfoUpdated(&old.OCP, &new.OCP)
+	}
+	return false
+}
+
+func ocpDistributionInfoUpdated(old, new *clusterv1beta1.OCPDistributionInfo) bool {
+	if len(old.AvailableUpdates) != len(new.AvailableUpdates) {
+		return true
+	}
+	if len(new.AvailableUpdates) > 1 {
+		sort.SliceStable(new.AvailableUpdates, func(i, j int) bool { return new.AvailableUpdates[i] < new.AvailableUpdates[j] })
```

Comment:
how about
```go
func ocpDistributionInfoUpdated(old, new *clusterv1beta1.OCPDistributionInfo) bool {
  sort.SliceStable(new.AvailableUpdates, func(i, j int) bool { return new.AvailableUpdates[i] < new.AvailableUpdates[j] })
   sort.SliceStable(new.VersionAvailableUpdates, func(i, j int) bool {			return new.VersionAvailableUpdates[i].Version &lt; new.VersionAvailableUpdates[j].Version		})
  sort.SliceStable(new.VersionHistory, func(i, j int) bool { return new.VersionHistory[i].Version < new.VersionHistory[j].Version })
  return !equality.Semantic.DeepEqual(old, new)
}
```

Reply:
yes, this looks like more clear. 
 I checked that sort.SliceStable can handle empty slice.
done.